components/Authentication/signup.jsx

```diff
@@ -60,46 +59,90 @@ const ShowText = styled.Text`
   padding-right: 15px;
 `;
 
+const ErrorText = styled.Text`
+  font-family: 'Montserrat_600SemiBold';
+  color: red;
+  font-size: 12px;
+`;
+
+const ViewFiller = styled.View`
+  height: 15px;
+`;
+
 const Signup = ({ navigation }) => {
   const [name, setName] = React.useState('');
+  const [validName, setValidName] = React.useState(false);
   const [email, setEmail] = React.useState('');
+  const [validEmail, setValidEmail] = React.useState(false);
   const [password, setPassword] = React.useState('');
+  const [validPassword, setValidPassword] = React.useState(false);
   const [hidePassword, setHidePassword] = React.useState(true);
+  const isFirstRunName = React.useRef(true);
+  const isFirstRunEmail = React.useRef(true);
+  const isFirstRunPassword = React.useRef(true);
+
+  // TODO: this regex isn't fully functional
+  function ValidateEmail(mail) {
+    // eslint-disable-next-line
+    const re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    if (re.test(mail)) {
+      return (true);
+    }
+    return (false);
+  }
+
+  React.useEffect(() => {
+    if (isFirstRunName.current) {
+      isFirstRunName.current = false;
+      return;
+    }
+    setValidName(!((name.length > 1)));
+  }, [name]);
+
+  React.useEffect(() => {
+    if (isFirstRunEmail.current) {
+      isFirstRunEmail.current = false;
+      return;
+    }
+    setValidEmail(!ValidateEmail(email));
+  }, [email]);
+
+  React.useEffect(() => {
+    if (isFirstRunPassword.current) {
+      isFirstRunPassword.current = false;
+      return;
+    }
+    setValidPassword(!(password.length > 5));
+  }, [password]);
```

Comment:
I don't quite understand this code, I didn't see your comment last night and idk where it went anymore haha.

Reply:
All good. It is basically just there to stop it from triggering the first time, since useEffect triggers on init of a state var so I didn't want all the invalid field text things to show when the user hasn't even clicked on the input lines. So after they click on the input, it will allow it to show the invalid text thing.

Reply:
Okay, that makes sense. I feel like there's probably an easier way to do this, but I can't think of one off the top of my head and I think its fine cause this works

Reply:
I thought so too. This was the only way I could figure out, it was more of an issue than I expected